### PR TITLE
[FIX] cannot create user due to required name.

### PR DESCRIPTION
**Pull request by NL66278**
Use @api.onchange to prevent error with required name when creating user.

### CODE REVIEW

**Comment by NL66278** [comment]
Solves: https://github.com/OCA/partner-contact/issues/112

File: partner_firstname/res_user.py [hunk 1]
```diff
@@ -17,14 +18,19 @@
...
+class ResUsers(models.Model):
+    """Extend res.users to be compatible with split name in res.partner."""
     _inherit = 'res.users'
 
+    @api.onchange('firstname', 'lastname')
+    def change_name(self):
+        names = [name for name in [self.firstname, self.lastname] if name]
+        self.name = ' '.join(names)
+
     def copy_data(self, cr, uid, _id, default=None, context=None):
```
Comment (pedrobaeza) [inline partner_firstname/res_user.py:34]: You can take the occasion to change this to new API.

**Comment by yvaucher** [comment]
:+1:

**Comment by NL66278** [comment]
@pedrobaeza Hello Pedro, I tried to make the changes as small as possible. The copy_data method itself in openerp/models.py has still the same interface as before.

**Comment by yvaucher** [comment]
I agree with @NL66278 to keep this diff small. API 8.0 on copy_data is out of context for a fix.

**Comment by pedrobaeza** [comment]
OK then :+1:

**Pull request merged**